e2e/tests/interchain_accounts/groups_test.go

```diff
@@ -0,0 +1,185 @@
+package interchain_accounts
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	grouptypes "github.com/cosmos/cosmos-sdk/x/group"
+	ibctest "github.com/strangelove-ventures/ibctest"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/test"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+	controllertypes "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/controller/types"
+	icatypes "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	ibctesting "github.com/cosmos/ibc-go/v6/testing"
+	simappparams "github.com/cosmos/ibc-go/v6/testing/simapp/params"
+)
+
+func TestInterchainAccountsGroupsTestSuite(t *testing.T) {
+	suite.Run(t, new(InterchainAccountsGroupsTestSuite))
+}
+
+type InterchainAccountsGroupsTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+func (s *InterchainAccountsGroupsTestSuite) QueryGroupPolicyAddress(ctx context.Context, chain ibc.Chain) string {
+	queryClient := s.GetChainGRCPClients(chain).GroupsQueryClient
+	res, err := queryClient.GroupPoliciesByGroup(ctx, &grouptypes.QueryGroupPoliciesByGroupRequest{
+		GroupId: 1, // always use the initial group id
+	})
+	s.Require().NoError(err)
+
+	return res.GroupPolicies[0].Address
+}
+
+func (s *InterchainAccountsGroupsTestSuite) TestInterchainAccountsGroupsIntegration_Success() {
+	t := s.T()
+	ctx := context.TODO()
+
+	var (
+		groupPolicyAddr   string
+		interchainAccAddr string
+		err               error
+	)
+
+	// setup relayers and connection-0 between two chains
+	// channel-0 is a transfer channel but it will not be used in this test case
+	relayer, _ := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	chainAAddress := chainAWallet.Bech32Address(chainA.Config().Bech32Prefix)
+
+	chainBWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+	chainBAddress := chainBWallet.Bech32Address(chainB.Config().Bech32Prefix)
+
+	t.Run("create group with new threshold decision policy", func(t *testing.T) {
+		members := []grouptypes.MemberRequest{
+			{
+				Address: chainAAddress,
+				Weight:  "1",
```

Comment:
extract to const?

Reply:
probably a good habit

Reply:
what is a threshold decision policy? What does the weight do?

Reply:
The `ThresholdDecisionPolicy` defines a weighted threshold that x amount of YES votes must meet in order to succeed. Each group member can carry a different weight which is accounted for in the tally. 

I've moved a bunch of fields to consts at the top of file and added godocs (some copy pasted from sdk).